### PR TITLE
Replace void cast with `[[maybe_unused]]`

### PR DESCRIPTION
**Pull request by fernandohv3279**
Replaced void cast with the more modern [[maybe_unused]].

Working towards fixing #17247

### CODE REVIEW

File: source/grid/tria.cc [hunk 1]
```diff
@@ -849,9 +849,9 @@ namespace internal
     else
 #endif
       {
-        (void)global_first_cell;
-        (void)global_num_cells;
-        (void)mpi_communicator;
+        // (void)global_first_cell;
+        // (void)global_num_cells;
+        // (void)mpi_communicator;
```
Comment (blaisb) [inline source/grid/tria.cc:854]: Instead of commenting these lines you can remove them :).
```suggestion
```

File: source/grid/tria.cc [hunk 2]
```diff
@@ -1064,8 +1064,8 @@ namespace internal
     else // if (mpisize > 1)
 #endif
       {
-        (void)global_first_cell;
-        (void)global_num_cells;
+        // (void)global_first_cell;
+        // (void)global_num_cells;
```
Comment (blaisb) [inline source/grid/tria.cc:1068]: Same here, you can delete this lines :) 
```suggestion
```

File: source/grid/tria.cc [hunk 3]
```diff
@@ -1064,8 +1064,8 @@ namespace internal
...
         (void)mpi_communicator;
```
Comment (blaisb) [inline source/grid/tria.cc:1069]: Can you verify if that statement is still necessary ? :)

File: source/grid/tria.cc (hunk 1 above)
Comment (bangerth) [reply source/grid/tria.cc:854]: By way of explanation, what @blaisb is saying is that there is no value in keeping these lines around: Why would anyone want to know that these lines were once part of the source code? So if there is no value in keeping them, let's just remove them!

File: source/grid/tria.cc (hunk 3 above)
Comment (fernandohv3279) [reply source/grid/tria.cc:1064]: Are you referring to line 1069?

Comment (tjhei) [reply source/grid/tria.cc:1064]: Yes, the   (void)mpi_communicator; line.

Comment (fernandohv3279) [reply source/grid/tria.cc:1064]: If that is the case, I didn't remove it because I only did the first 5, another person did line 1069 I think :)

Comment (blaisb) [reply source/grid/tria.cc:1064]: Perfect :). My bad sorry

**Review by blaisb — APPROVED** [review]

**Review by tamiko — APPROVED** [review]

**Comment by bangerth** [comment]
This currently fails with the following error:
```
Invalid author 'fernando' without firstname and lastname

For possible solutions, consult the following page:
      https://github.com/dealii/dealii/wiki/Commit-authorship
```
Can you consult that page for how to fix the issue?

**Comment by fernandohv3279** [comment]
> This currently fails with the following error:
> 
> ```
> Invalid author 'fernando' without firstname and lastname
> 
> For possible solutions, consult the following page:
>       https://github.com/dealii/dealii/wiki/Commit-authorship
> ```
> 
> Can you consult that page for how to fix the issue?

I believe I fixed that. I have no clue what the new problem is though.

**Comment by masterleinad** [comment]
> I believe I fixed that. I have no clue what the new problem is though.

You'll have to squash all your commits (or fix the author in all your commits).

**Comment by fernandohv3279** [comment]
I think the indent is correct now, but there are some other errors

**Comment by marcfehling** [comment]
Superseded by #17617.

**Pull request closed without merging**